### DISCUSSION TITLE
Conflict free multiple users editing

### DISCUSSION THREAD

**Discussion by 99ansh** [tinacms/tinacms#4639] [Ideas]
For self hosted TinaCMS I have observed that when multiple users are trying to edit the same page, the content gets overridden by the last save. There should be a way to introduce locking (page level/field level) for conflict free editing experience.

**Comment by bradystroud** [comment]
If you have multiple users, branching might be the way to go
https://tina.io/docs/tina-cloud/branching/

Here is a good video on branching 
https://youtu.be/XvL3pFiYaVw

**Reply by 99ansh** [reply]
Updated description as this is request is for self hosted
